[doc][rdt] Add the limitations of rdt (#58063)
ray-project/ray: doc/source/ray-core/direct-transport.rst

   * Any unexpected system bugs

For NIXL:

> * Due to an issue with our implementation of memory deregistration, we currently do not support repeated transfers of tensors that share the same memory space but belong to different objects. We will fix this problem soon.

Reviewer: Can you give like 2 small examples - 
1. sending 2 lists of tensors that overlap
[a, b, c], [c, d, e]

2. sending the same tensor twice

> * Due to an issue with our implementation of memory deregistration, we currently do not support repeated transfers of tensors that share the same memory space but belong to different objects. We will fix this problem soon.

Reviewer 2: We might as well do the error detection in the same PR so we can provide a code sample showing what error to expect.

Also, I think this assumes too much system knowledge from the user.

> Due to an issue with our implementation of memory deregistration,

-> "Due to a known issue"

>  repeated transfers of tensors that share the same memory space but belong to different objects.

Technically this is possible, but you have to make sure to free the first object before the second. It's probably clearer with code examples, like dhyey said.

   * Any unexpected system bugs

For NIXL:

> * Due to an issue with our implementation of memory deregistration, we currently do not support repeated transfers of tensors that share the same memory space but belong to different objects before the object in the first transfer is freed from Ray's gpu object store. Here are some examples of what will not work:

Reviewer: changing to just
```suggestion
* Due to a known issue, we currently do not support repeated transfers of tensors that share the same memory space but belong to different objects before the object in the first transfer is freed from Ray's gpu object store. Here are some examples of what will not work:
```
